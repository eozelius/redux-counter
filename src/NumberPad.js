import React from 'react'
import CalculatorButton from './CalculatorButton'

const NumberPad = (props) => {
  const numbers = [9,8,7,6,5,4,3,2,1,0, '.']
  return(
    <div className='numberPadContainer'>
      { numbers.map((n, i) => <CalculatorButton onClick={props.handleNumber} display={n} key={i} />) }
    </div>
  )
}

export default NumberPad
