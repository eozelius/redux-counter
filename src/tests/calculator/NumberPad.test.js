import React from 'react'
import NumberPad from '../../NumberPad'

describe(NumberPad, () => {
  it('mounts properly', () => {
    const pad = shallow(<NumberPad />)
    expect(pad).toMatchSnapshot()
  })

  it('handles a number click ', () => {
    const fn = jest.fn()
    const pad = mount(<NumberPad handleNumber={fn} />)
    pad.find('.calcBtn').first().simulate('click')
    expect(fn).toHaveBeenCalled()
  })
})
