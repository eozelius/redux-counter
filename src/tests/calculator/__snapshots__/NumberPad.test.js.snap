// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberPad mounts properly 1`] = `
<div
  className="numberPadContainer"
>
  <CalculatorButton
    display={9}
    key="0"
  />
  <CalculatorButton
    display={8}
    key="1"
  />
  <CalculatorButton
    display={7}
    key="2"
  />
  <CalculatorButton
    display={6}
    key="3"
  />
  <CalculatorButton
    display={5}
    key="4"
  />
  <CalculatorButton
    display={4}
    key="5"
  />
  <CalculatorButton
    display={3}
    key="6"
  />
  <CalculatorButton
    display={2}
    key="7"
  />
  <CalculatorButton
    display={1}
    key="8"
  />
  <CalculatorButton
    display={0}
    key="9"
  />
  <CalculatorButton
    display="."
    key="10"
  />
  <CalculatorButton
    className="executeResult"
    display="="
  />
</div>
`;
